app/src/main/java/adapters/ProfileFragmentAdapter.java

```diff
@@ -0,0 +1,31 @@
+package adapters;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.viewpager2.adapter.FragmentStateAdapter;
+
+import fragments.profileFragments.MyGroupsFragment;
+import fragments.profileFragments.MyPostsFragment;
+
+public class ProfileFragmentAdapter extends FragmentStateAdapter {
+    final static int NUMBER_OF_TABS = 2;
+
+    public ProfileFragmentAdapter(@NonNull Fragment fragment) {
+        super(fragment);
+    }
+
+    @NonNull
+    @Override
+    public Fragment createFragment(int position) {
+        switch (position) {
+            case 1:
```

Comment:
This is probably worth adding a definition for each position.
 i.e. private static final GROUP_FRAGMENT_SELECTED(I'm guessing here...) = 1;
 private static final MY_POSTS_FRAGMENT_SELECTED = 2;
 ...
 switch (position) {
 case GROUP_FRAGMENT_SELECTED:
...
case MY_POSTS_FRAGMENT_SELECTED:
...
default:
LOG_ERROR(WTF HAPPENED);